ivadomed/ivadomed
# Fix dropout_rate key in models.py

## Checklist

#### GitHub


#### PR contents


## Description
This PR fixes issue #926 where the key for dropout was different between the config files and the expected keys in classes of `models.py`.
- As discussed, I updated the key in all models and classes of `models.py` from `drop_rate` to `dropout_rate`.
- I updated the documentation of the default Unet model with default values for `dropout`, `bn_momentum` and `depth`. Those values are applied by default by the `config_manager` from the `config_default.json` file when the key is missing from the config file.

I tested the following architectures: `default_model` (Unet), the `FiLMedUnet`, the `Modified3DUnet` and `HeMISUnet`.

### Questions:
@charleygros, the default value from the `config_default.json` for dropout is `0.3`, whereas the default values in `models.py` are `0`, `0.4` or `0.6`. I don't think that is an issue i.e if the model is called via the config file, the default will be 0.3 but if the model is called directly, the default value will be `0`, `0.4` or `0.6` depending on the model as documented in the docstrings. Let me know what you think.

Also the initialization of `FiLMedUnet(Unet)` looks a bit strange to me:
https://github.com/ivadomed/ivadomed/blob/d71a1c4711d03da7c61a473f571134b262684073/ivadomed/models.py#L528-L530
The `super().__init__` with Unet uses default values, I'm not sure but it would make more sense to me to use to values define by `def __init__`.
ex:
```
def __init__(self, in_channel=1, out_channel=1, depth=3, dropout_rate=0.4,
             bn_momentum=0.1, n_metadata=None, film_layers=None, is_2d=True, n_filters=64, **kwargs):
    super().__init__(in_channel=in_channel, out_channel=out_channel, depth=depth, dropout_rate=dropout_rate, bn_momentum=bn_momentum)
```
At the end of the day, it does not change the behavior and the right values from the config file are applied in the model, so maybe I'm being picky here.

### To test the PR:
- Run a training with a dropout value different from 0.3 in the config_file.
- print the model [after this line](https://github.com/ivadomed/ivadomed/blob/d71a1c4711d03da7c61a473f571134b262684073/ivadomed/training.py#L101) and check the value of dropout in the model.

## Linked issues
Fixes #926
